pkg/utils/comparators/versions.go

```diff
@@ -0,0 +1,59 @@
+package comparators
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// CompareVersions will return one of:
+// -1 : if lhs > rhs
+// 0  : if lhs == rhs
+// 1  : if rhs > lhs
+func CompareVersions(lhs, rhs string) int {
+	lVersions := buildVersionArray(lhs)
+	rVersions := buildVersionArray(rhs)
+
+	lLen := len(lVersions)
+	rLen := len(rVersions)
+
+	for i := 0; i < lLen && i < rLen; i++ {
+		if lVersions[i] > rVersions[i] {
+			return -1
+		}
+
+		if lVersions[i] < rVersions[i] {
+			return 1
+		}
+	}
+
+	// check if lhs is a more specific version number (aka newer)
+	if lLen > rLen {
+		return -1
+	}
+
+	// check if rhs is a more specific version number
+	if lLen < rLen {
+		return 1
+	}
+
+	// versions are exactly the same
+	return 0
+}
+
+func buildVersionArray(version string) []int {
```

Comment:
Consider using an existing library https://godoc.org/golang.org/x/mod/semver

Reply:
es doesn't use semantic versioning...
`In this package, semantic version strings must begin with a leading "v", as in "v1.0.0`

also, i feel like this is simple enough that it doesn't warrant pulling in a library for.